In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns #importo la libreria
import plotly

In [3]:
empleos = pd.read_excel("datos/mujeres/mujeres.xlsx") #cargo los datos

In [4]:
empleos.genero.unique()   

array(['femenino', 'mujer trans', 'femenino|masculino', 'no binario',
       'femenino|no binario'], dtype=object)

In [4]:
empleos.columns

Index(['autor', 'etiqueta', 'descripcion', 'ocupacion', 'genero', 'nacAño',
       'nacLugar', 'nacLat', 'nacLon', 'nacPais', 'morAño', 'morLugar',
       'morLat', 'morLon', 'morPais', 'mncarsBib', 'qid', 'wikidataURL',
       'viaf', 'viafURL', 'bne', 'bneURL', 'mncars', 'mncarsCatWeb',
       'wikipedias', 'wk_ES', 'wkPREF'],
      dtype='object')

# Representación por países 

In [5]:
empleos.nacPais.unique()

array([nan, 'Finlandia', 'España', 'Polonia', 'Estados Unidos',
       'Argentina', 'Serbia', 'Reino Unido', 'Alemania', 'Italia',
       'República Popular China', 'Bélgica', 'Suecia', 'Suiza', 'Austria',
       'Israel', 'Francia', 'Bosnia y Herzegovina', 'México', 'Chile',
       'Sudáfrica', 'Irak', 'Canadá', 'Portugal', 'Brasil', 'Egipto',
       'Australia', 'Países Bajos', 'Nueva Zelanda', 'Cuba', 'Bahamas',
       'Rusia', 'Japón', 'Colombia', 'Perú', 'Venezuela', 'Lituania',
       'Noruega', 'Eslovenia', 'ocupación aliada en Alemania',
       'Imperio romano', 'Argelia', 'Hungría', 'Marruecos', 'Islandia',
       'Dinamarca', 'Ucrania', 'Paraguay', 'India', 'Kenia',
       'Reino de los Países Bajos', 'Grecia', 'República Checa',
       'Croacia', 'Uruguay', 'Túnez', 'Letonia', 'Corea del Sur',
       'Panamá', 'Rumania', 'Indonesia', 'Irlanda', 'Georgia', 'Liberia',
       'Vietnam del Sur', 'Turquía', 'Senegal', 'Checoslovaquia',
       'Mozambique', 'Líbano', 'Pakistán', '

In [19]:
empleos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10101 entries, 0 to 10100
Data columns (total 27 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   autor         10101 non-null  object 
 1   etiqueta      10101 non-null  object 
 2   descripcion   9342 non-null   object 
 3   ocupacion     9432 non-null   object 
 4   genero        10101 non-null  object 
 5   nacAño        9617 non-null   float64
 6   nacLugar      6847 non-null   object 
 7   nacLat        6826 non-null   float64
 8   nacLon        6826 non-null   float64
 9   nacPais       6837 non-null   object 
 10  morAño        3036 non-null   float64
 11  morLugar      2330 non-null   object 
 12  morLat        2322 non-null   float64
 13  morLon        2322 non-null   float64
 14  morPais       2325 non-null   object 
 15  mncarsBib     10101 non-null  object 
 16  qid           10101 non-null  object 
 17  wikidataURL   10101 non-null  object 
 18  viaf          9974 non-nul

Hay muchas entradas en las que no aparece el país. Solo 6837 entradas tienen valor en el campo nacPais

# PIB

In [10]:
url = 'https://es.wikipedia.org/wiki/Anexo:Pa%C3%ADses_por_PIB_(nominal)'
t = pd.read_html(url)
len(t)

7

In [54]:
t[4]

Pos. País (o territorio) PIB nominal (millones  de USD)   Año
0      NaN               Mundo                     79 580 000  2017
1      1.0      Estados Unidos                     19 360 000  2017
2      NaN       Unión Europea                     16 520 000  2017
3      2.0               China                     11 940 000  2017
4      3.0               Japón                      4 884 000  2017
..     ...                 ...                            ...   ...
214  190.0            Kiribati                            186  2017
215    NaN        Anguila (RU)                           1754  2009
216    NaN  Islas Malvinas[7]​                           1645  2007
217  191.0              Tuvalu                             40  2017
218    NaN           Niue (NZ)                           1001  2003

[219 rows x 4 columns]

In [64]:
# creo una tabla excel a partir de los datos de la wikipedia
pib = pd.read_excel('PIB_2021.xlsx')
pib

País PIB (nominal millones de USD)
0           Estados Unidos                    22 997 500
1                    China                    17 725 821
2                    Japón                     4 938 780
3                 Alemania                     4 220 591
4                    India                     3 037 901
5              Reino Unido                     3 020 243
6                  Francia                     2 935 716
7                   Italia                     2 105 462
8                   Canadá                     1 990 566
9            Corea del Sur                     1 797 081
10                   Rusia                     1 775 036
11               Australia                     1 733 884
12                  Brasil                     1 611 864
13                  España                     1 421 979
14                  México                     1 292 298
15               Indonesia                     1 183 053
16                    Irán                     1 081 383
17            Países Bajos                     1 015 563
18          Arabia Saudita                       842 363
19                   Suiza                       812 730
20                 Turquía                       809 641
21                 Polonia                       673 508
22                  Suecia                       625 575
23                 Bélgica                       599 232
24               Tailandia                       505 666
25                 Irlanda                       498 261
26                 Noruega                       481 988
27               Argentina                       480 986
28                  Israel                       480 904
29                 Austria                       476 797
30                  Egipto                       438 348
31                 Nigeria                       429 976
32               Sudáfrica                       417 836
33  Emiratos Árabes Unidos                       410 158
34               Dinamarca                       395 528
35               Filipinas                       393 440
36                Singapur                       374 147
37                 Vietnam                       366 290
38                 Malasia                       357 422
39               Bangladés                       354 574

# Defino dos funciones: quitar_espacios_importes y quitar_espacios_nombres_pais. La primera función divide en palabras y luego une las palabras sin espacios. La segunda función elimina los espacios en blanco al  y al final. 

In [11]:
pib = pd.read_excel('datos/pib/PIB_2021.xlsx')
def quitar_espacios_importes(x):
    l = x.split()
    d = ''.join(x for x in l)    
    return int(d)


def quitar_espacios_nombres_pais(x):
    l = x.strip()    
    return l


pib['PIB (float)'] = pib['PIB (nominal millones de USD)'].map(quitar_espacios_importes)
pib.País = pib.País.map(quitar_espacios_nombres_pais)
pib

País PIB (nominal millones de USD)  PIB (float)
0           Estados Unidos                    22 997 500     22997500
1                    China                    17 725 821     17725821
2                    Japón                     4 938 780      4938780
3                 Alemania                     4 220 591      4220591
4                    India                     3 037 901      3037901
5              Reino Unido                     3 020 243      3020243
6                  Francia                     2 935 716      2935716
7                   Italia                     2 105 462      2105462
8                   Canadá                     1 990 566      1990566
9            Corea del Sur                     1 797 081      1797081
10                   Rusia                     1 775 036      1775036
11               Australia                     1 733 884      1733884
12                  Brasil                     1 611 864      1611864
13                  España                     1 421 979      1421979
14                  México                     1 292 298      1292298
15               Indonesia                     1 183 053      1183053
16                    Irán                     1 081 383      1081383
17            Países Bajos                     1 015 563      1015563
18          Arabia Saudita                       842 363       842363
19                   Suiza                       812 730       812730
20                 Turquía                       809 641       809641
21                 Polonia                       673 508       673508
22                  Suecia                       625 575       625575
23                 Bélgica                       599 232       599232
24               Tailandia                       505 666       505666
25                 Irlanda                       498 261       498261
26                 Noruega                       481 988       481988
27               Argentina                       480 986       480986
28                  Israel                       480 904       480904
29                 Austria                       476 797       476797
30                  Egipto                       438 348       438348
31                 Nigeria                       429 976       429976
32               Sudáfrica                       417 836       417836
33  Emiratos Árabes Unidos                       410 158       410158
34               Dinamarca                       395 528       395528
35               Filipinas                       393 440       393440
36                Singapur                       374 147       374147
37                 Vietnam                       366 290       366290
38                 Malasia                       357 422       357422
39               Bangladés                       354 574       354574

In [12]:
## Cruzo los datos con los paises del museo

In [13]:
paises

Estados Unidos    1266
España            1023
Alemania           701
Francia            545
Reino Unido        386
                  ... 
Uzbekistán           1
Sri Lanka            1
Mónaco               1
Afganistán           1
Eritrea              1
Name: nacPais, Length: 127, dtype: int64

In [14]:
pib.País.value_counts() #cuento los valores

Estados Unidos            1
China                     1
Suecia                    1
Bélgica                   1
Tailandia                 1
Irlanda                   1
Noruega                   1
Argentina                 1
Israel                    1
Austria                   1
Egipto                    1
Nigeria                   1
Sudáfrica                 1
Emiratos Árabes Unidos    1
Dinamarca                 1
Filipinas                 1
Singapur                  1
Vietnam                   1
Malasia                   1
Polonia                   1
Turquía                   1
Suiza                     1
Corea del Sur             1
Japón                     1
Alemania                  1
India                     1
Reino Unido               1
Francia                   1
Italia                    1
Canadá                    1
Rusia                     1
Arabia Saudita            1
Australia                 1
Brasil                    1
España                    1
México              

#  Realizo una fusión entre dos DataFrames: paises y pib, utilizando el índice del DataFrame paises y la columna 'País' del DataFrame pib.  Creo un dataframe llamado pib_numero_mujers.

In [15]:
pib_numero_mujers = pd.merge(paises, pib, left_index = True, right_on = 'País' )
pib_numero_mujers

nacPais                    País PIB (nominal millones de USD)  PIB (float)
0      1266          Estados Unidos                    22 997 500     22997500
13     1023                  España                     1 421 979      1421979
3       701                Alemania                     4 220 591      4220591
6       545                 Francia                     2 935 716      2935716
5       386             Reino Unido                     3 020 243      3020243
7       252                  Italia                     2 105 462      2105462
29      161                 Austria                       476 797       476797
8       160                  Canadá                     1 990 566      1990566
19      159                   Suiza                       812 730       812730
27      132               Argentina                       480 986       480986
21      129                 Polonia                       673 508       673508
17      126            Países Bajos                     1 015 563      1015563
12      117                  Brasil                     1 611 864      1611864
22       94                  Suecia                       625 575       625575
14       93                  México                     1 292 298      1292298
2        87                   Japón                     4 938 780      4938780
23       84                 Bélgica                       599 232       599232
10       79                   Rusia                     1 775 036      1775036
11       60               Australia                     1 733 884      1733884
28       59                  Israel                       480 904       480904
26       36                 Noruega                       481 988       481988
34       32               Dinamarca                       395 528       395528
25       28                 Irlanda                       498 261       498261
4        27                   India                     3 037 901      3037901
32       21               Sudáfrica                       417 836       417836
9        18           Corea del Sur                     1 797 081      1797081
16       15                    Irán                     1 081 383      1081383
30       14                  Egipto                       438 348       438348
20       11                 Turquía                       809 641       809641
37        5                 Vietnam                       366 290       366290
35        4               Filipinas                       393 440       393440
31        4                 Nigeria                       429 976       429976
15        4               Indonesia                     1 183 053      1183053
39        2               Bangladés                       354 574       354574
36        2                Singapur                       374 147       374147
38        1                 Malasia                       357 422       357422
24        1               Tailandia                       505 666       505666
18        1          Arabia Saudita                       842 363       842363
33        1  Emiratos Árabes Unidos                       410 158       410158

# Creo un gráfico de dispersión (scatter plot) utilizando los datos del DataFrame pib_numero_mujers. Las coordenadas x corresponden a la columna 'nacPais', las coordenadas y corresponden a la columna 'PIB (float)', y las etiquetas de texto se toman de la columna 'País'.

In [16]:
import plotly.graph_objs as go

from plotly.offline import init_notebook_mode, iplot
init_notebook_mode(connected=True)

In [18]:
trace = go.Scatter(
    x = pib_numero_mujers.nacPais,
    y = pib_numero_mujers['PIB (float)'],
    mode = 'markers',
    text = pib_numero_mujers.País,
    
)

data = [trace]

layout = dict(
            title='Número de artistas mujeres por país ',
            yaxis = dict(
                title = "PIB (en USD)"
            ),
            xaxis = dict(
                title = "Número de mujeres artistas"
            )
         )
fig = go.Figure(data=data, layout=layout)
iplot(fig)



No parece que el número de artistas aumente si aumenta el pib del país.